d20/tests/test_GameMaster.py

```diff
@@ -318,3 +341,1637 @@ def testhandleAddObject4(self, FileObjectInit):
+        " specified in options"
+    args.temporary = '/tmp/d20-test'
+
+
+def testBackStoryFactLoad1(monkeypatch):
+    mockBSF = mock.Mock()
+    mock1 = mock.Mock()
+
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerBackStories",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerNPCs",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerPlayers",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerScreens",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.resolveBackStoryFacts",
+                        mockBSF)
+
+    args = args_ex
+    args.backstory_facts = """
+    facts:
+        - name: BulkAnalyzeFact
+          arguments:
+            directory: /path/to/files
+            recursive: False
+            enable: True
+        - name: VTDownloadFact
+          arguments:
+            vt_api_key: your_api_key
+            filehash: hash_to_lookup_and_download_for_analysis
+            enable: False
+    """
+
+    gm = GameMaster(options=args)
+    mockBSF.assert_called()
+    gm.cleanup()
+
+
+def testBackStoryFactPathLoad2(monkeypatch):
+    mockBSF = mock.Mock()
+    mock1 = mock.Mock()
+
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerBackStories",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerNPCs",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerPlayers",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.GameMaster.registerScreens",
+                        mock1)
+    monkeypatch.setattr("d20.Manual.GameMaster.resolveBackStoryFacts",
+                        mockBSF)
+
+    tf = tempfile.NamedTemporaryFile(delete=False)
+    tf.write(b"""
+        facts:
+            - name: BulkAnalyzeFact
+              arguments:
+                directory: /path/to/files
+                recursive: False
+                enable: True
+            - name: VTDownloadFact
+              arguments:
+                vt_api_key: your_api_key
+                filehash: hash_to_lookup_and_download_for_analysis
+                enable: False
+    """)
+
+    tf.close()
+    args = args_ex
+    args.backstory_facts_path = tf.name
+
+    GameMaster(options=args)
+    mockBSF.assert_called()
+    os.remove(tf.name)
+
+
+def testFileOpenException(monkeypatch):
+    exception_mock = mock.Mock(return_value=Exception)
+    mock1 = mock.Mock()
+    tf = tempfile.NamedTemporaryFile(delete=False)
+    tf.close()
+    args = args_ex
+    args.file = tf.name
+
+    with monkeypatch.context() as m:
+        m.setattr('builtins.open', exception_mock)
+        m.setattr('d20.Manual.GameMaster.LOGGER.exception', mock1)
+        with pytest.raises(SystemExit):
```

Comment:
I see we don't have anything specific that is outputted for some of these errors besides what is in the logs, so you may want to use `caplog` fixture in pytest to get the output log and make sure there is some text from the function about the expected error (obviously this may be a bit more fragile than a custom exception though..).  I also saw you used caplog below, so it might be good in some other places.